### PR TITLE
[NEW] Video conference

### PR DESCRIPTION
**Pull request by philipbrito**
This PR brings video call support into Rocket.Chat Android app. 
This PR also update the Coroutines version to the stable one.

**Known issues:**

- Not working on Android < M (more info: https://github.com/jitsi/jitsi-meet/pull/3967).

**Known improvements to do:**

- Video is showing the conferencing title.
- Better use of coroutines scope here.

Notes:

- Would be great to have a better managing of the Jitsi SDK version used within the app. Hopefully Jitsi maintainers will make it more clear very soon - I should admit not finding this information easily (https://github.com/jitsi/jitsi-meet/issues/3987)

### CODE REVIEW

**Comment by rockneverdies55** [comment]
Exciting :+1:

File: app/src/main/java/chat/rocket/android/app/AppLifecycleObserver.kt [hunk 1]
```diff
@@ -33,7 +29,7 @@ class AppLifecycleObserver @Inject constructor(
     }
 
     private fun changeTemporaryStatus(userStatus: UserStatus) {
-        launch {
+        GlobalScope.launch {
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/app/AppLifecycleObserver.kt:32]: We should use a specific scope here. To be improved in a new PR (see https://github.com/RocketChat/Rocket.Chat.Android/issues/2140)

File: app/src/main/java/chat/rocket/android/chatroom/adapter/ChatRoomAdapter.kt [hunk 2]
```diff
@@ -6,12 +6,7 @@ import android.view.ViewGroup
 import androidx.recyclerview.widget.RecyclerView
 import chat.rocket.android.R
 import chat.rocket.android.chatroom.presentation.ChatRoomNavigator
-import chat.rocket.android.chatroom.uimodel.AttachmentUiModel
-import chat.rocket.android.chatroom.uimodel.BaseUiModel
-import chat.rocket.android.chatroom.uimodel.MessageReplyUiModel
-import chat.rocket.android.chatroom.uimodel.MessageUiModel
-import chat.rocket.android.chatroom.uimodel.UrlPreviewUiModel
-import chat.rocket.android.chatroom.uimodel.toViewType
+import chat.rocket.android.chatroom.uimodel.*
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatroom/adapter/ChatRoomAdapter.kt]: Dont'use wildcards!

File: app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt [hunk 3]
```diff
@@ -12,17 +11,13 @@ import android.os.Bundle
 import android.os.Handler
 import android.provider.MediaStore
 import android.text.SpannableStringBuilder
-import android.view.KeyEvent
-import android.view.LayoutInflater
-import android.view.Menu
-import android.view.MenuItem
-import android.view.View
-import android.view.ViewGroup
+import android.view.*
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt]: Don't use wildcards!

File: app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt [hunk 4]
```diff
@@ -34,12 +29,7 @@ import androidx.recyclerview.widget.RecyclerView
 import chat.rocket.android.R
 import chat.rocket.android.analytics.AnalyticsManager
 import chat.rocket.android.analytics.event.ScreenViewEvent
-import chat.rocket.android.chatroom.adapter.ChatRoomAdapter
-import chat.rocket.android.chatroom.adapter.CommandSuggestionsAdapter
-import chat.rocket.android.chatroom.adapter.EmojiSuggestionsAdapter
-import chat.rocket.android.chatroom.adapter.PEOPLE
-import chat.rocket.android.chatroom.adapter.PeopleSuggestionsAdapter
-import chat.rocket.android.chatroom.adapter.RoomSuggestionsAdapter
+import chat.rocket.android.chatroom.adapter.*
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt]: Don't use wildcards!

File: app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt [hunk 5]
```diff
@@ -53,35 +43,19 @@ import chat.rocket.android.chatroom.uimodel.suggestion.PeopleSuggestionUiModel
...
 import chat.rocket.android.draw.main.ui.DrawingActivity
-import chat.rocket.android.emoji.ComposerEditText
-import chat.rocket.android.emoji.Emoji
-import chat.rocket.android.emoji.EmojiKeyboardListener
-import chat.rocket.android.emoji.EmojiKeyboardPopup
-import chat.rocket.android.emoji.EmojiParser
-import chat.rocket.android.emoji.EmojiPickerPopup
-import chat.rocket.android.emoji.EmojiReactionListener
-import chat.rocket.android.emoji.internal.GlideApp
+import chat.rocket.android.emoji.*
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt]: Don't use wildcards!

File: app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt [hunk 6]
```diff
@@ -53,35 +43,19 @@ import chat.rocket.android.chatroom.uimodel.suggestion.PeopleSuggestionUiModel
...
-import chat.rocket.android.util.extensions.fadeIn
-import chat.rocket.android.util.extensions.fadeOut
-import chat.rocket.android.util.extensions.hideKeyboard
-import chat.rocket.android.util.extensions.inflate
-import chat.rocket.android.util.extensions.isNotNullNorEmpty
-import chat.rocket.android.util.extensions.rotateBy
-import chat.rocket.android.util.extensions.showToast
-import chat.rocket.android.util.extensions.textContent
-import chat.rocket.android.util.extensions.ui
+import chat.rocket.android.util.extensions.*
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt]: Don't use wildcards!

File: app/src/main/java/chat/rocket/android/chatroom/adapter/ChatRoomAdapter.kt (hunk 2 above)
Comment (philipbrito) [reply app/src/main/java/chat/rocket/android/chatroom/adapter/ChatRoomAdapter.kt]: Solved.

File: app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt (hunk 3 above)
Comment (philipbrito) [reply app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt]: Solved.

File: app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt (hunk 4 above)
Comment (philipbrito) [reply app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt]: Solved.

File: app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt (hunk 5 above)
Comment (philipbrito) [reply app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt]: Solved.

File: app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt (hunk 6 above)
Comment (philipbrito) [reply app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt]: Solved.

**Pull request merged**